libsolidity/analysis/TypeChecker.h

```diff
@@ -164,6 +164,9 @@ class TypeChecker: private ASTConstVisitor
 
 	ContractDefinition const* m_scope = nullptr;
 
+	/// Pointer to the function we're currently in
```

Comment:
Why is it important that we are currently inside that function?

Reply:
I think this stems from a misunderstanding from my side. The issue description says

> This is a rather specific case - you have to call the non-implemented base function and override the function at the same time.

I thought this means it only happens when you call it from within the function that overrides which is what I am detecting here.

@bshastry newly found test case shows that the problem is more general than this.